dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java

```diff
@@ -66,17 +78,31 @@ public void clear() {
 
     @Override
     public Object get(int index) {
-        return null;
+        return this._array[index];
     }
 
     @Override
     public Object set(int index, Object element) {
-        return null;
+        if (index > maxSize) {
+            return null;
+        }
+        else {
+            Object previousElement = this._array[index];
+            this._array[index] = element;
+            return previousElement;
+        }
     }
 
     @Override
     public void add(int index, Object element) {
-
+        if (this.currentSize != this.maxSize) {
+            for (int i=0; i < this.currentSize-(index);i++){
+                _array[this.currentSize-i] = _array[this.currentSize-i-1];
```

Comment:
Nice work! It's safer to check `this.currentSize < this.maxSize` but your way definitely works.